.github/workflows/stream-diff.yml

```diff
@@ -16,20 +16,39 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      - name: Genericize new metadata
-        run: |
-          mkdir -p new/streams
-          for stream in streams/*.json; do
-              echo "${stream}"
-              ci/genericize-stream.py "${stream}" > "new/${stream}"
-          done
-      - name: Genericize old metadata
+      - name: Genericize metadata
         run: |
+          # Cache data from PR head
+          mkdir head
+          cp -r streams ci/genericize-stream.py head
+
+          # Switch to base
           git checkout "${GITHUB_BASE_REF}"
-          mkdir -p old/streams
+
+          # Get the numerically latest fedora-coreos-stream-generator tag
+          generator_tag=$(
+              curl -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' https://api.github.com/repos/coreos/fedora-coreos-stream-generator/git/matching-refs/tags/ |
+              jq -r .[].ref |
+              sed s:^refs/tags/:: |
+              sort -V |
+              tail -n 1)
+
+          # Genericize metadata
+          mkdir -p {old,new}/streams
           for stream in streams/*.json; do
-              echo "${stream}"
-              ci/genericize-stream.py "${stream}" > "old/${stream}"
+              echo "${stream}, old"
+              # If the stream metadata has changed in the head commit,
+              # pretend the old metadata used the latest tag of
+              # fedora-coreos-stream-generator.
+              generator_arg=
+              if ! cmp -s "${stream}" "head/${stream}"; then
+                  echo "  Setting generator to fedora-coreos-stream-generator ${generator_tag}"
+                  generator_arg="-g ${generator_tag}"
+              fi
+              head/genericize-stream.py ${generator_arg} "${stream}" > "old/${stream}"
+
+              echo "${stream}, new"
+              head/genericize-stream.py "head/${stream}" > "new/${stream}"
```

Comment:
It feels like if the stream metadata hasn't changed (i.e. `cmp -s "${stream}" "head/${stream}"` tells us they are the same) then we don't need to run `genericize-stream.py` at all on either old or new and we can just copy the files as is into `"old/${stream}"` and `"new/${stream}"` and continue the loop early.

so if we just make it a separate check early in the loop then continue if that's the case then we can just genericize the old `streams/*.json` with `-g ${generator_tag}` unconditionally:

```
head/genericize-stream.py -g ${generator_tag} "${stream}" > "old/${stream}"
```

Reply:
Yeah, good call.  That's a good cleanup.